stackit/internal/services/redis/credential/datasource.go

```diff
@@ -159,6 +161,11 @@ func (r *credentialDataSource) Read(ctx context.Context, req datasource.ReadRequ
 
 	recordSetResp, err := r.client.GetCredentials(ctx, projectId, instanceId, credentialId).Execute()
 	if err != nil {
+		oapiErr, ok := err.(*oapierror.GenericOpenAPIError) //nolint:errorlint //complaining that error.As should be used to catch wrapped errors, but this error should not be wrapped
+		if ok && oapiErr.StatusCode == http.StatusNotFound {
+			resp.State.RemoveResource(ctx)
+			return
```

Comment:
all credential datasources are returning before throwing error, is that on purpose?